pkg/ruler/notifier/notifier_config.go

```diff
@@ -0,0 +1,99 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package notifier
+
+import (
+	"flag"
+	"fmt"
+	"hash/fnv"
+
+	"github.com/grafana/dskit/crypto/tls"
+	"github.com/grafana/dskit/flagext"
+	"gopkg.in/yaml.v3"
+
+	"github.com/grafana/mimir/pkg/util"
+)
+
+var DefaultAlertmanagerClientConfig = AlertmanagerClientConfig{
+	NotifierConfig: Config{
+		OAuth2: OAuth2Config{
+			EndpointParams: flagext.NewLimitsMap[string](nil),
+		},
+	},
+}
+
+type AlertmanagerClientConfig struct {
+	AlertmanagerURL string `yaml:"alertmanager_url" doc:"nocli"`
```

Comment:
I believe this also should have the CLI params and be visible in the `limits` help block, same as the other pattern discussed